### DISCUSSION TITLE
Can't execute Winget via Group Action -> Powershell

### DISCUSSION THREAD

**Discussion by RTechSn** [Ylianst/MeshCentral#6555] [Issues] [answered]
Hi. 
I'm trying to automate installation of certain apps, but i can't get 'winget' command to work via group action. 
If i remote to the host and open Powershell, winget command is executing file.
But when i do 'Run commands' -> 'Windows Powershell' -> 'Commands from Text box', i get this error: 
```
winget : The term 'winget' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
 the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ winget
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (winget:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
Run commands completed.
```

**Comment by si458** [comment]
the is a bug with the run commands, where if you select 'run as user' it doesnt run as a user 😨 
it will run as the NT SYSTEM user like the agent does 
you can verify this my simply running `whoami`
could this be your issue?

**Reply by si458** [reply]
also, ive never used winget but from what ive seen, it installs winget into the current user and not the system
so you might need to install winget using the terminal as the NT SYSTEM user, then it would work?

**Reply by si458** [reply]
https://github.com/microsoft/winget-cli/discussions/962 you might not be able to use it as the SYSTEM user, and you need to use the current user instead,
so in this case it wont work because of a BUG we have the the `run as user` option, sorry

**Reply by RTechSn** [reply]
Yes, looks like this is it.
And there is no point to run Winget as SYSTEM, as the app i'm trying to install should be in AppData folder.
Can you send me the the link to the issue? Thanks!

**Reply by si458** [reply, marked as answer]
here is a few links to the `run as user` bug in meshcentral
https://github.com/Ylianst/MeshCentral/issues/4949
https://github.com/Ylianst/MeshCentral/issues/6429 
https://github.com/Ylianst/MeshCentral/issues/6430
